shared.js

```diff
@@ -125,6 +125,8 @@ var Shared = module.exports = {
 
 		this.sandbox.run(engine, 4000, code, function(result) {
 			var reply;
+			if (result.text) result.text = result.text.replace(/\r/g, '\n');
```

Comment:
this will cause extra newlines since most `\r`s are going to be paired with `\n`s - I think you may want `result.test.replace(/\r?\n?/g, '\n')`?